src/System.Security.Cryptography.Cng/tests/PropertyTests.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Cng.Tests
+{
+    public static class PropertyTests
+    {
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public static void SetPropertyZeroLengthCornerCase()
+        {
+           using (CngKey key = CngKey.Import(TestData.Key1, CngKeyBlobFormat.GenericPublicBlob))
+           {
+               const string propertyName = "CustomZeroLengthProperty";
+               CngProperty p = new CngProperty(propertyName, new byte[0], CngPropertyOptions.CustomProperty);
+               key.SetProperty(p);
+
+               CngProperty p2 = key.GetProperty(propertyName, CngPropertyOptions.CustomProperty);
+               Assert.Equal(propertyName, p2.Name);
+               Assert.Equal(CngPropertyOptions.CustomProperty, p2.Options);
+
+               // This one is odd. CNG keys can have properties with zero length but CngKey.GetProperty() transforms this into null.
+               Assert.Equal(null, p2.GetValue());
+
+               Assert.Throws<CryptographicException>(() => key.GetProperty("DOES NOT EXIST", CngPropertyOptions.CustomProperty));
```

Comment:
This feels like a distinct test (GetNotSpecifiedProperty / GetProperty_NotSpecified)